FHEM/00_SIGNALduino.pm

```diff
@@ -2801,6 +2801,41 @@ sub SIGNALduino_Parse($$$$@) {
     return ;
   }
 
+  ### advance message checks ###
+  # --> abort for invalid messages
+  # --> detection of errors in firmware or incorrect transfer in the project
+
+  $hash->{logMethod}->($name, AttrVal($name,'noMsgVerbose',5), "$name: Parse, Msg found: $rmsg");
+  if ($rmsg =~ /^MU/)
```

Comment:
Die Prüfungen der Daten erfolgt bereits ab Zeile 2858 und ff.

Da der Feher in Zeile 2669 ist (SIGNALduino_Parse_MC), haben wir in SIGNALduino_Split_Message wohl kein Problem mit fehlerhaften Daten.

Zum einen könnten wir die Prüfung in Zeile 2868 verbessern. Andererseits liegt das eigentliche Problem aber in SIGNALduino_Parse_MC, weil wir dieser sub so ziemlich alles mit signal_parts unterjubeln können und keinerlei Prüfung der Inhalte erfolgt.

Reply:
Bitte versteife dich nicht nur auf die Parse_MC. Es betrifft alle Parse Routinen.

Der Grundgedanke ist, das wir bei einer falschen bzw. Fehlerhaften  Nachricht nicht die Parse aufrufen müssten. Ein „vorzeitiger“ Abbruch reduziert die Verarbeitung und Warnungen. Ein „stiller“ Abbruch wäre aber nicht optimal. Wenn, dann bitte eine Logausgabe um den Fehlerfall auch zu registrieren.